src/background.js

```diff
@@ -0,0 +1,38 @@
+function runShowMeTheCode(tabId, tab) {
+    function isGitHub(tab) {
+        return urlMatches(tab, /https:\/\/github\.com\/.*/)
+    };
+
+    function isBlobView(tab) {
+        return urlMatches(tab, /.*\/blob\/.*/);
+    };
+
+    function pullRequestFileView(tab) {
+        return urlMatches(tab, /.*\/pull\/.*\/files/);
+    };
+
+    function urlMatches(tab, expression) {
+        return tab.url && tab.url.match(expression);
+    }
+
+    function insertCSS(fileName) {
+        chrome.tabs.insertCSS(tabId, {file: fileName});
+    };
+
+    if (isGitHub(tab)) {
+        if (isBlobView(tab) || pullRequestFileView(tab)) {
+            insertCSS('showMeTheCode.css');
+        } else {
+            insertCSS('disableShowMeTheCode.css');
```

Comment:
How can we achieve that? Page is not being reloaded and our CSS stays loaded
18 gru 2015 6:12 PM "Łukasz Nowak" <notifications@github.com> napisał(a):

> In src/background.js
> <https://github.com/Viridovix/show-me-the-code-github/pull/32#discussion_r48046976>
> :
>
> > +        return urlMatches(tab, /.*\/pull\/.*\/files/);
> > +    };
> > +
> > +    function urlMatches(tab, expression) {
> > +        return tab.url && tab.url.match(expression);
> > +    }
> > +
> > +    function insertCSS(fileName) {
> > +        chrome.tabs.insertCSS(tabId, {file: fileName});
> > +    };
> > +
> > +    if (isGitHub(tab)) {
> > +        if (isBlobView(tab) || pullRequestFileView(tab)) {
> > +            insertCSS('showMeTheCode.css');
> > +        } else {
> > +            insertCSS('disableShowMeTheCode.css');
>
> Can't we use original style?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Viridovix/show-me-the-code-github/pull/32/files#r48046976>
> .
>